db/models/article.models.js

```diff
@@ -1,15 +1,28 @@
 const db = require("./../connection")
 
-exports.retrieveAllArticles = () => {
-    return db
+exports.retrieveAllArticles = (topic) => {
+    if(topic) {
+        return db
+        .query(`SELECT articles.author, articles.title, articles.topic, articles.article_id, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id) AS comment_count
+        FROM articles
+        FULL OUTER JOIN comments ON articles.article_id = comments.article_id
+        WHERE topic = $1
+        GROUP BY articles.author, articles.title, articles.topic, articles.article_id, articles.created_at, articles.votes, articles.article_img_url
+        ORDER BY created_at DESC`, [topic])
+        .then(({ rows }) => {
+            return rows
+        })
+    } else {
+        return db
```

Comment:
This really should be building up a query string based on the topic, not two entire db queries.